### ISSUE TITLE
Hot loader not working correctly

### ISSUE THREAD

**Issue by n8sabes** [jquintozamora/react-typescript-webpack2-cssModules-postCSS#1]
The current configuration is causing the entire page to reload after editing / webpack update. Hot loader should only update changed nodes and styles. Is there something special about the use of TypeScript compiler that prevents targeted updates via the hot loader? As an example check out how [RSB](https://github.com/kriasoft/react-static-boilerplate) handles edits and hot loader page updates.

**Comment by jquintozamora** [comment]
Hi @n8sabes , 
I have tried to emulate this issue, but I see when I edit a .ts file or .css file (using Chrome) then HMR is changing only the affected component as you can see in the image:
![image](https://cloud.githubusercontent.com/assets/12530031/25993275/5fba21e4-3701-11e7-996d-8d5850091982.png)

Just to make sure we both are in the same page, please could you provide more information about the issue?

Thanks!

**Comment by jquintozamora** [comment]
Closing the issue, as I am not able to reproduce and there is no more information.

**Comment by egorshulga** [comment]
I confirm the issue.
This happens to me during the second change being applied.
![image](https://user-images.githubusercontent.com/6253488/28463472-3c1e35c2-6e29-11e7-91ee-bc76dfa215fd.png)

**Comment by jquintozamora** [comment]
Hi @egorshulga ,

Thanks for your comment.

Did you change index.ts second time? 

Tkae into account that HMR is working under App.tsx but not in index.ts. The reason is because index.ts is where the code related to HMR configuration is placed. 

Let me know if further questions / issues.

**Comment by egorshulga** [comment]
@jquintozamora, I have changed App.tsx both times.

**Comment by egorshulga** [comment]
@jquintozamora, besides, why have you used so complicated HMR configuration? Why simple "--hot --inline" switches are not suitable for the case?

**Comment by egorshulga** [comment]
So I switched from `ts-loader` to `awesome-typescript-loader` and the problem has gone.
I have no clue what kind of a bug it was (I'm not an expert or anything).

@jquintozamora, thanks for the quick reply. 

Hope, it could help anyone with the issue.

**Comment by jquintozamora** [comment]
Hi @egorshulga ,
Thank you.

I just solved the issue.
I just published the changes and I have published a new version (https://github.com/jquintozamora/react-typescript-webpack2-cssModules-postCSS/releases/tag/v2.1). 

The problem was the way in which is referenced module.hot.accept. That is the right way:

// Hot Module Replacement API
if (module.hot) {
  module.hot.accept(() => {
    const NextApp = require<{default: typeof App}>("./containers/App/App").default;
    render(
      <AppContainer>
        <NextApp />
      </AppContainer>
      ,
      reactContainer
    );
  });
}


About ts-loader, I prefer using that one, because awesome-typescript-loader gave me few problems in the past when using declarations within TypeScript.

Thanks!
